### PR TITLE
Added multiple and numbered %s support for search engines

### LINKED ISSUE #987: Only first %s is replaced with search query
*   Brief description of the problem:
  Using search engines with more than one encounters of `%s` doesn't work correctly -- only the first one is replaced with search query and the other remain unchanged.
E.g.: Let a search engine be `https://www.oxfordlearnersdictionaries.com/definition/english/%s?q=%s` and a search query `benevolence` than resulting URL is
`https://www.oxfordlearnersdictionaries.com/definition/english/benevolence?q=%s`.
Note: due to features of Oxford Learner's Dictionary's implementation some queries such as `hello?q=%s` will result in loading proper page and some will not. 

*   Steps to reproduce:

    1. Add a search engine with two entries of `%s`
    2. Try to search something
    3. Check the resulting URL

*   Tridactyl version (`:version`):
1.13.3-1.13.3
*   Firefox version (Top right menu > Help > About Firefox):
Mozilla Firefox 60.1.0 (ESR)
*   URL of the website the bug happens on:
Tested on https://www.oxfordlearnersdictionaries.com/
*   Config (in a new tab, run `:viewconfig`, copy the url and paste it somewhere like gist.github.com):
https://gist.github.com/vanzef/ebed49be44149285f5079232b87decfd

### PR DESCRIPTION
**Pull request by vanzef**
closes #987

### CODE REVIEW

File: src/url_util.ts [hunk 1]
```diff
@@ -363,9 +363,20 @@ export function interpolateSearchItem(urlPattern: URL, query: string): URL {
         query = encodeURIComponent(query)
     }
 
+    queryWords = query.split(" ")
```
Comment (bovine3dom) [inline src/url_util.ts]: Did you test this? Travis says that this hasn't been defined amongst other things. I think you're probably missing a few lets.

**Comment by glacambre** [comment]
Thank you for taking the time to make Tridactyl better! I tried your patch and found why it didn't work, the following patch should fix it:
```
diff --git a/src/url_util.ts b/src/url_util.ts
index 8abca4d..68631ec 100644
--- a/src/url_util.ts
+++ b/src/url_util.ts
@@ -354,6 +354,8 @@ export function graftUrlPath(url: URL, newTail: string, level: number) {
 export function interpolateSearchItem(urlPattern: URL, query: string): URL {
     const hasInterpolationPoint = urlPattern.href.includes("%s")
 
+    let queryWords = query.split(" ")
+
     // percent-encode if theres a %s in the query string, or if we're apppending
     // and there's a query string
     if (
@@ -361,15 +363,14 @@ export function interpolateSearchItem(urlPattern: URL, query: string): URL {
         urlPattern.search !== ""
     ) {
         query = encodeURIComponent(query)
+        queryWords = queryWords.map(w => encodeURIComponent(w))
     }
 
-    queryWords = query.split(" ")
-
     // replace or append as needed
     if (hasInterpolationPoint) {
-        resultingURL = new URL (urlPattern.href.replace(/%s\d+/g, function(x) {
-            const index = (x.slice(2) as number) - 1
-            if (index > queryWords.length()) {
+        let resultingURL = new URL (urlPattern.href.replace(/%s\d+/g, function(x) {
+            const index = parseInt(x.slice(2)) - 1
+            if (index >= queryWords.length) {
                 return ""
             }
 ```

I can't push it to your branch so you'll need to do that yourself (or maybe bovine3dom can do it). Again, thank you very much :)

**Comment by vanzef** [comment]
Sorry for the mess. I've never written in JavaScript/TypeScript before.

I'll apply @glacambre's patch right away and push it.

File: src/url_util.ts (hunk 1 above)
Comment (vanzef) [reply src/url_util.ts]: Fixed by @glacambre

**Comment by bovine3dom** [comment]
Thanks. Merged.

We should document this _somewhere_ but I couldn't decide exactly where so merged it anyway.

Probably in config.ts above searchurls?

**Pull request merged**